.github/workflows/auto-deprecate.yml

```diff
@@ -0,0 +1,31 @@
+name: NPM Auto Deprecate
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 1 * * *'
+
+jobs:
+  auto-deprecate:
+    name: NPM Auto Deprecate
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@v2
+      - name: Use Node.js v16
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - name: Restore CI Cache
+        uses: actions/cache@v2.1.6
+        id: cache-restore
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-16-${{ hashFiles('**/package-lock.json') }}
+      - name: Install Dependencies if Cache Miss
+        if: ${{ !steps.cache-restore.outputs.cache-hit }}
+        run: npm ci
+      - name: Deprecate versions
+        run: npx npm-deprecate --name "*dev*" --package "discord.js"
```

Comment:
Would a --no-install be desirable here?

Reply:
It shouldn't matter because I added it as a devdep. 

As as aside, `--no-install` is deprecated, see: https://docs.npmjs.com/cli/v7/commands/npx#compatibility-with-older-npx-versions

Reply:
Wasn't aware --no was already implemented, nice.

> It shouldn't matter because I added it as a devdep.

Even more reason to do it, and with npm's replacement for npx, it does prompt if the command can't be found, meaning a stalled workflow if the dep fails to install the command properly, technically that's on your package, but its a thought.

Reply:
> it does prompt if the command can't be found

I've actually never seen it prompt in CI actually... maybe they check for that too?

Reply:
I haven't gotten a chance to run npm exec / npx in CI yet, so cannot confirm or deny with that, just remember that the old npx did not prompt at all. 

It does seem like CI detection is performed though in order to prevent this behavior, just not sure what the result is
> Interactive mode is not supported in CI environments when standard input is a TTY, to prevent hangs.

Reply:
The result is a warning + installing + executing from what I saw 👀

Reply:
Added the flag and also changed it to `npm exec` since that's the new way. Tested the command how it's written now locally and it passes args properly (I made it so adding `--verbose` / `-v` will print the options it receives)